packages/react-router/lib/router/router.ts

```diff
@@ -2052,22 +2016,9 @@ export function createRouter(init: RouterInit): Router {
       loaderResults,
       pendingActionResult,
       revalidatingFetchers,
-      fetcherResults,
-      activeDeferreds
+      fetcherResults
     );
 
-    // Wire up subscribers to update loaderData as promises settle
-    activeDeferreds.forEach((deferredData, routeId) => {
-      deferredData.subscribe((aborted) => {
-        // Note: No need to updateState here since the TrackedPromise on
-        // loaderData is stable across resolve/reject
-        // Remove this instance if we were aborted or if promises have settled
-        if (aborted || deferredData.done) {
-          activeDeferreds.delete(routeId);
-        }
-      });
-    });
```

Comment:
No longer needed - promises will settle and be handled in the UI by application code